vizro-core/src/vizro/models/_navigation/_accordion.py

```diff
@@ -40,32 +44,19 @@ def _create_accordion_buttons(self, accordion_pages):
                 href=page["relative_path"],
             )
             for page in dash.page_registry.values()
-            for accordion_page in accordion_pages
-            if accordion_page == page["module"] and page["module"] != MODULE_PAGE_404
+            if page["name"] in pages
         ]
 
-    def _create_accordion_item(self, accordion_buttons, title=ACCORDION_TITLE):
+    def _create_accordion_item(self, accordion_buttons, title=ACCORDION_DEFAULT_TITLE):
+        """Creates an accordion item for each sub-group of pages."""
         return dbc.AccordionItem(
             children=accordion_buttons,
             title=title.upper(),
             class_name="accordion_item",
         )
 
-    def _create_default_accordion(self):
-        accordion_buttons = [
-            dbc.Button(
-                children=[page["name"]],
-                key=page["relative_path"],
-                className="accordion-item-button",
-                href=page["relative_path"],
-            )
-            for page in dash.page_registry.values()
-            if page["module"] != MODULE_PAGE_404
-        ]
-
-        accordion_items = [self._create_accordion_item(accordion_buttons=accordion_buttons)]
-
-        # Don't create accordion navigation if there is only one page and one accordion item
+    def _get_accordion_container(self, accordion_items, accordion_buttons):
+        # Return no container if there is only one page in the dashboard
```

Comment:
Could it happen that `len(accordion_buttons)` is equal to 1 and that `len(accordion_items)` isn't? 
It looks like we can replace the `if` statement below with the following:
`if len(accordion_buttons) == 1:`

Reply:
That check wouldn't suffice! For example, if you provide the navigation as below - you would have ` len(accordion_buttons) = 1` while `len(accordion_items) = 2`. In this case, I don't want to return the Accordion. So the check that both accordion_buttons and accordion_items are of length 1 are required to ensure that only one page exists in the dashboard :

```
 navigation=vm.Navigation(
        pages={
            "Analysis": ["Variable Analysis"],
            "Summary": ["Continent Summary"],
        }
    ),
```